dockstore-integration-testing/src/test/java/io/dockstore/client/cli/WorkflowIT.java

```diff
@@ -145,6 +150,37 @@ public void testTargettedRefresh() throws ApiException {
         assertTrue("should find 4 valid versions for bitbucket workflow, found : " + refreshBitbucket.getWorkflowVersions().stream()
                         .filter(WorkflowVersion::isValid).count(),
                 refreshBitbucket.getWorkflowVersions().stream().filter(WorkflowVersion::isValid).count() == 4);
+
+
+    }
+
+    @Test
+    public void testUnableToLaunchUnpublishedToolCreds() throws IOException {
+        String toolpath = SourceControl.GITHUB.toString() + "/DockstoreTestUser2/md5sum-checker/test";
+        final CommonTestUtilities.TestingPostgres testingPostgres = getTestingPostgres();
+        testingPostgres.runUpdateStatement("update enduser set isadmin = 't' where username = 'DockstoreTestUser2';");
+        final ApiClient webClient = getWebClient();
+        WorkflowsApi workflowApi = new WorkflowsApi(webClient);
+        Workflow workflow = workflowApi
+            .manualRegister(SourceControl.GITHUB.getFriendlyName(), "DockstoreTestUser2/md5sum-checker", "checker_workflow_wrapping_workflow.cwl",
+                "test", "cwl", null);
+        Workflow refresh = workflowApi.refresh(workflow.getId());
+
+        Assert.assertTrue("workflow is already published for some reason", !refresh.isIsPublished());
+
+        // should be able to launch properly with correct credentials even though the workflow is not published
+        FileUtils.writeStringToFile(new File("md5sum.input"), "foo" , StandardCharsets.UTF_8);
+        systemExit.expectSystemExitWithStatus(0);
+        Client.main(new String[] { "--config", ResourceHelpers.resourceFilePath("config_file2.txt"), "workflow", "launch", "--entry", toolpath, "--json" , ResourceHelpers.resourceFilePath("md5sum-wrapper-tool.json") ,  "--script" });
+
+        // should not be able to launch properly with incorrect credentials
+        systemExit.expectSystemExitWithStatus(Client.ENTRY_NOT_FOUND);
+        Client.main(new String[] { "--config", ResourceHelpers.resourceFilePath("config_file.txt"), "workflow", "launch", "--entry", toolpath, "--json" , ResourceHelpers.resourceFilePath("md5sum-wrapper-tool.json") ,  "--script" });
+    }
+
+    @Test
+    public void testAbletoLaunchUnpublishedToolWithCredentials(){
```

Comment:
Empty test, looks like this test is merged together with the one above to avoid duplicate manual register and refresh?  If so, change the other test name to reflect that it is able to launch an unpublished workflow with the right credentials and that is isn't able to do so without credentials.  On an unrelated note, doesn't seem checkstyle is working for tests.

Reply:
Yup, merged for rate limit purposes. 
Accidentally left this in. Checkstyle is more relaxed for tests, we can tighten it over time